src/fidocadj/primitives/PrimitiveAdvText.java

```diff
@@ -465,199 +490,182 @@ public int getDistanceToPoint(int px, int py)
-            six=((Integer)pd.parameter).intValue();
-        } else {
-            System.out.println("Warning: unexpected parameter!"+pd);
-        }
+     */
+    private void addParameter(List<ParameterDescription> list, Object parameter,
+            String descriptionKey)
+    {
+        ParameterDescription pd = new ParameterDescription();
+        pd.parameter = parameter;
+        pd.description = Globals.messages.getString(descriptionKey);
+        list.add(pd);
+    }
+    
+    /**
+     Set the control parameters of the given primitive.
+     This method is specular to getControls().
 
-        pd=(ParameterDescription)v.get(i);
-        ++i;
-        if (pd.parameter instanceof Integer) {
-            siy=((Integer)pd.parameter).intValue();
-        } else {
-            System.out.println("Warning: unexpected parameter!"+pd);
-        }
+     @param parameters a vector of ParameterDescription containing each control
+     parameter.
 
-        pd=(ParameterDescription)v.get(i);
-        ++i;
-        if (pd.parameter instanceof Integer) {
-            o=((Integer)pd.parameter).intValue();
-        } else {
-            System.out.println("Warning: unexpected parameter!"+pd);
-        }
+     @return the next index in v to be scanned (if needed) after the
+     execution of this function.
+     */
+    @Override
+    public int setControls(List<ParameterDescription> parameters)
+    {
+        changed = true;
+        recalcSize = true;
 
-        pd=(ParameterDescription)v.get(i++);
-        if (pd.parameter instanceof Boolean) {
-            sty = ((Boolean)pd.parameter).booleanValue()?
-                sty | TEXT_MIRRORED:
-                sty & (~TEXT_MIRRORED);
-        } else {
-            System.out.println("Warning: unexpected parameter!"+pd);
-        }
+        int i = 0;
 
-        pd=(ParameterDescription)v.get(i++);
-        if (pd.parameter instanceof Boolean) {
-            sty = ((Boolean)pd.parameter).booleanValue() ?
-                sty | TEXT_ITALIC:
-                sty & (~TEXT_ITALIC);
-        } else {
-            System.out.println("Warning: unexpected parameter!"+pd);
-        }
+        for (ParameterDescription pd : parameters) {
+            Object param = pd.parameter;
 
-        pd=(ParameterDescription)v.get(i++);
-        if (pd.parameter instanceof Boolean) {
-            sty= ((Boolean)pd.parameter).booleanValue() ?
-                sty | TEXT_BOLD:
-                sty & (~TEXT_BOLD);
-        } else {
-            System.out.println("Warning: unexpected parameter!"+pd);
-        }
+            if (param instanceof String) {
+                txt = (String) param;
+            } else {
+                if (param instanceof LayerInfo) {
+                    setLayer(((LayerInfo) param).getLayer());
+                } else {
+                    if (param instanceof Integer) {
+                        switch (i) {
+                            case 2 ->
```

Comment:
There is an error here? Is the -> correct? This does not look as a lambda?

Reply:
> There is an error here? Is the -> correct? This does not look as a lambda?

Feature introduced in Java 12 as part of a new syntax for switch statements. This new syntax, called "switch expression," allows the switch construct to be used more concisely and with less boilerplate code compared to the traditional syntax.

[See this reference](https://blogs.oracle.com/javamagazine/post/new-switch-expressions-in-java-12)